batchTesting.m

```diff
@@ -0,0 +1,121 @@
+function batchTesting(parallel)
+% function batchTesting(parallel)
+%
+% Execute all the xunit Test functions of the toolbox
+% in batch mode.
+%
+% Inputs:
+%
+% parallel - true for parallel execution
+%
+% Example:
+%
+% batchTesting(1)
+%
+% author: hugo.oliveira@utas.edu.au
+%
+
+% Copyright (C) 2020, Australian Ocean Data Network (AODN) and Integrated
+% Marine Observing System (IMOS).
+%
+% This program is free software: you can redistribute it and/or modify
+% it under the terms of the GNU General Public License as published by
+% the Free Software Foundation version 3 of the License.
+%
+% This program is distributed in the hope that it will be useful,
+% but WITHOUT ANY WARRANTY; without even the implied warranty of
+% MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+% GNU General Public License for more details.
+%
+% You should have received a copy of the GNU General Public License
+% along with this program.
+% If not, see <https://www.gnu.org/licenses/gpl-3.0.en.html>.
+%
+
+root_folder = pwd;
+addpath([root_folder filesep 'Util' filesep 'Path'])
+setToolboxPaths(root_folder)
+
+try
+    results = runAllTests(parallel)  %do not supress report output
+catch
+    error('Tests failed')
+end
+
+failed = {'    Tests failed:\n'};
+c = 2;
+
+for k = 1:length(results)
+    ktest = results(k);
+
+    if ~ktest.Passed
+        fparam = extract_param(ktest.Name);
+        fnames = fieldnames(fparam);
+        fnames = fnames(2:end);
+        fstr = [repmat(' ',1,length(failed{1})) fparam.testname ':\n'];
+        lfstr = length(fstr)
+        for kk = 1:length(fnames)
+            fstr = [fstr repmat(' ',1,lfstr) fnames{kk} '->' fparam.(fnames{kk}) '\n'];
+        end
+        fstr = [fstr '\n'];
+        failed{c} = fstr;
+        c = c + 1;
```

Comment:
I'm a bit confused by all this loop. Would you mind adding some comments?

Reply:
I included some comments and improve the linting that was lacking.

THe for loop just extract the results of failed tests and store them as a string for pretty print later.
The results of the tests in the matlab xunit testing are ugly and hard to read - I want something more straight forward when doing the testing and checking the results.